spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/functions/IcebergVersionFunctionImpl.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.functions;
+
+import org.apache.iceberg.IcebergBuild;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.functions.ScalarFunction;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * {@link org.apache.spark.sql.connector.catalog.functions.BoundFunction} that implements {@link
+ * org.apache.iceberg.spark.functions.IcebergVersionFunction} by returning the short version string,
+ * e.g. "0.14.0" or "0.15.0-SNAPSHOT".
+ */
+public class IcebergVersionFunctionImpl implements ScalarFunction<UTF8String> {
```

Comment:
Why does this need to be public? I think this could easily be a private static class inside of `IcebergVersionFunction`.

Reply:
I made it public so I could test that it resolves to the right class type. But it could be at the least nested within the `IcebergVersionFunction` (which is how I had it before). I can update that.

Reply:
And then I'll see about the `public` or not after moving it back.

Reply:
So it can be a nested subclass (known).

It can also be package-private, but making it `private` makes the `invoke` method not accessible.

When invoke is `public static` and the subclass is private static I get the error message:

```
interface org.apache.spark.sql.catalyst.expressions.objects.InvokeLike cannot access a member of class org.apache.iceberg.spark.functions.IcebergVersionFunction$IcebergVersionFunctionImpl with modifiers "public static"
java.lang.IllegalAccessException: interface org.apache.spark.sql.catalyst.expressions.objects.InvokeLike cannot access a member of class org.apache.iceberg.spark.functions.IcebergVersionFunction$IcebergVersionFunctionImpl with modifiers "public static"
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
	at java.base/java.lang.reflect.Method.invoke(Method.java:558)
```

Making invoke anything other than `public static` gives an error about no function `invoke` defined  (indicating it needs to be public). So the best it seems I can do is package-private as a subclass.